classes/quiz_settings.php

```diff
@@ -402,4 +438,49 @@ private function split_keys(string $keys) : array {
         }
         return $keys;
     }
+
+    /**
+     * Try and get a file in the user draft filearea by itemid.
+     *
+     * @param string $itemid Item ID of the file.
+     * @return stored_file|null Returns null if no file is found.
+     *
+     * @throws \coding_exception
+     */
+    private function get_current_user_draft_file(string $itemid) : ?stored_file { // @codingStandardsIgnoreLine
+        global $USER;
+        $context = context_user::instance($USER->id);
+        $fs = get_file_storage();
+        $files = $fs->get_area_files($context->id, 'user', 'draft', $itemid);
+        foreach ($files as $file) {
+            // Get first non empty file. Should only be one.
+            if ($file->get_filesize() > 0) {
+                $configfile = $file;
+            }
+        }
+        if (!empty($configfile)) {
+            return $configfile;
+        }
+        return null;
+    }
+
+    private function save_filemanager_draftarea() : bool {
+        global $CFG;
+
+        // During plugin installation file_save_draft_area_files causes an exception that user id=0 is not found.
```

Comment:
Do you know why this method gest called During plugin installation  ?

Reply:
During the install.php there is a function quizaccess_seb_create_missing_settings();

The lib.php:75 function creates settings for all quizzes where there are none.

Reply:
but should it return empty $draftitemid on istall?

Reply:
No files were present during the install so it is not modifying the default value.

Out of this scope I asked why the settings were generated during install at all. Can we modify the access rule constructor to not throw an exception when settings are not found?

Reply:
Sure. If you think it will make the code better, then go ahead!

Reply:
New issue created https://github.com/catalyst/moodle-quizaccess_seb/issues/88